src/ruby/puppetserver-lib/puppet/server/ast_compiler.rb

```diff
@@ -24,26 +24,79 @@ def self.compile(compile_options)
       end
 
       def self.compile_ast(code, compile_options)
+        # TODO: move requires to optimal place and guard against FOSS not having bolt lib code
+        # CODEREVIEW: where should these requires be, and what should happen if they are not found?
+        require 'bolt'
```

Comment:
If we rescue any load errors when running Bolt so that it is optional, is this endpoint still usable?

My understanding when this endpoint was originally implemented was that Bolt was the only current consumer (or soon to be consumer), but it wasn't necessarily only for Bolt consumption (ie in the future we may open it up to non-Bolt use cases).

Reply:
I think so, it would roughly be:
```
if (using bolt)
  require 'bolt' 
  ... use the code in this PR...
else
  ..use the code that was here before
```
I can easily do that, i just didnt know if there was a usecase outside of bolt. I do know that there is very little value for plans in PE without supporting bolt types.